### PR TITLE
Issue/13326 fix threat found redirection

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #13326

With the introduction of the Scan screen, we can introduce support for redirection to Scan screen from the notifications. Previously, the app would redirect the user to a webview.

Merge instructions:
1. Make sure [FluxC's PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1859) is merged first
2. Update FluxC tag
2. Remove "Not ready for merge label

To test:
Prerequisities:
- disable Scan feature flag
1. Go to Notifications
2. Click on a "Jetpack Scan detected a possible security threat" item
3. Click on "Site's Scan Results" span
4. Notice a webview is shown
---------------------------
Prerequisities:
- enable Scan feature flag
1. Go to Notifications
2. Click on a "Jetpack Scan detected a possible security threat" item
3. Click on "Site's Scan Results" span
4. Notice Scan screen is shown
5. Run the test again, but make sure that you have selected a different site (SiteB) in the app than the one (SiteA) on which the threat was found -> make sure the Scan screen is shown for SiteA

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java [hunk 1]
```diff
@@ -428,6 +428,18 @@ public void showPostActivity(long siteId, long postId) {
         ReaderActivityLauncher.showReaderPostDetail(this, siteId, postId);
     }
 
+     public void showScanActivityForSite(long siteId) {
+         SiteModel site = mSiteStore.getSiteBySiteId(siteId);
+         if (site == null) {
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java]: Suggestion (💡): What if the duplicate with the toast and comments is reduced like that below:
```
     SiteModel site = getSiteOrToast(siteId);
     if (site != null) {
         ActivityLauncher.viewScan(this, site);
     }

    @Nullable
    private SiteModel getSiteOrToast(long siteId) {
        SiteModel site = mSiteStore.getSiteBySiteId(siteId);
        if (site == null) {
            // One way the site can be null: new site created, receive a notification from this site,
            // but the site list is not yet updated in the app.
            ToastUtils.showToast(this, R.string.blog_not_found);
        }
        return site;
    }
```

**Review by ParaskP7 — APPROVED** [review]
👋 @malinajirka  !

I have reviewed and tested this PR as per the instruction, good job! 🌟 

I have left only one suggestion (💡 ) for you. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply my suggestion. However, feel free to ignore them and merge the PR yourself.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java (hunk 1 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java]: Good call ;)! Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/13941/commits/16f0af7d80dc9dece4aecc3f29ad6de43161f692

**Comment by malinajirka** [comment]
Thanks for the review @ParaskP7! ;) I've implemented the suggested improvement. It's ready for another round. Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java]: 🙏

**Review by ParaskP7 — APPROVED** [review]
LGTM!

**Pull request merged**